### PR TITLE
Add auto deactivation test

### PR DESCRIPTION
**Pull request by rajivnathan**
e2e tests for the new auto deactivation controller.

Summary of the changes:
- Refactored some of the code to reuse functions related to creating and manipulating tiers
- Added a new test for verifying that auto deactivation triggers user deactivation

Paired with https://github.com/codeready-toolchain/host-operator/pull/281

### CODE REVIEW

File: test/e2e/user_management_test.go [hunk 1]
```diff
@@ -68,6 +69,35 @@ func (s *userManagementTestSuite) TestUserDeactivation() {
...
+
+		// Let's create a tier with deactivation enabled
+		deactivationTier := CreateNSTemplateTier(t, ctx, hostAwait, "deactivation-tier", DeactivationTimeoutDays(tierDeactivationPeriod))
+
+		mur := MoveUserToTier(t, hostAwait, userSignup.Spec.Username, *deactivationTier)
+
+		err := s.hostAwait.Client.Get(context.TODO(), types.NamespacedName{
+			Namespace: mur.Namespace,
+			Name:      mur.Name,
+		}, mur)
```
Comment (alexeykazakov) [inline test/e2e/user_management_test.go:86]: Why do you need to get Mur again here? Is something wrong with the mur you got from `MoveUserToTier()`?

File: test/e2e/user_management_test.go [hunk 2]
```diff
@@ -68,6 +69,35 @@ func (s *userManagementTestSuite) TestUserDeactivation() {
...
+		// We cannot wait days for deactivation so for the purposes of the e2e tests we use a hack to change the provisioned time to a time far enough
+		// in the past to trigger auto deactivation. Subtracting the tier deactivation period from the current time and setting this as the provisioned
+		// time should cause the deactivation controller to reconcile and see the mur is ready for deactivation.
+		tierDeactivationDuration := time.Duration(tierDeactivationPeriod*24) * time.Hour
+		mur.Status.ProvisionedTime = &metav1.Time{Time: time.Now().Add(-tierDeactivationDuration)}
+		err = s.hostAwait.Client.Status().Update(context.TODO(), mur)
+		require.NoError(s.T(), err)
+		s.T().Logf("masteruserrecord '%s' provisioned time adjusted", mur.Name)
+
+		err = s.hostAwait.WaitUntilMasterUserRecordDeleted(mur.Name)
```
Comment (alexeykazakov) [inline test/e2e/user_management_test.go:98]: Also check the usersignup.Spec.Deactivated and Status.

**Review by alexeykazakov — COMMENTED** [review]
Looks good! Just couple of comments/questions.

File: test/e2e/user_management_test.go (hunk 1 above)
Comment (rajivnathan) [reply test/e2e/user_management_test.go]: Oops, that's a mistake. Forgot to remove that after refactoring the code. Fixed in https://github.com/codeready-toolchain/toolchain-e2e/pull/162/commits/284053b155da49bcbd952b32e2ee8f5da4ba211f

**Review by alexeykazakov — APPROVED** [review]

File: test/e2e/user_management_test.go (hunk 2 above)
Comment (rajivnathan) [reply test/e2e/user_management_test.go:102]: Done

**Review by MatousJobanek — COMMENTED** [review]
Nice and elegant solution :+1:

**Pull request merged**